tests/test_E4_verification.py

```diff
@@ -160,3 +165,101 @@ def test_detect_excessive_excess_generation_in_bus_several_busses_two_warnings(c
+    soc_series = pd.Series(data=data, index=index)
+
+    storage = "storage_01"
+    # Add the SoC time series to the result JSON nested-dict
+    dict_values = {ENERGY_STORAGE: {storage: {TIMESERIES_SOC: soc_series},}}
+    # Test for the function's behavior with the current case
+    with caplog.at_level(logging.WARNING):
+        E4.verify_state_of_charge(dict_values=dict_values)
+    assert (
+        f"SoC of {storage} in this time-step is greater than 1. This is a physically impossible value!"
+        not in caplog.text
+    ), f"A WARNING message is logged even though the SoC values are all below 1."
+    assert (
+        f"SoC of {storage} in this time-step is less than 0. This is a physically impossible value!"
+        not in caplog.text
+    ), f"A WARNING message is logged even though the SoC values are all above 0."
+
+
+def test_verify_state_of_charge_soc_below_zero(caplog):
+    """Test case: The SoC value are all below 0.00 and thus, physically infeasible
+    Test if the WARNING log messages are logged"""
+
+    # Create a list of floating-point numbers less than 0.00
+    list_floats = [-1.23, -0.05, -1.05]
+    data = np.array(list_floats)
+
+    # Generate a date-time index
+    index = pd.date_range(start="1/1/2018", periods=3, tz="Asia/Kolkata")
+
+    # Create a SoC time series from above floats and date-time data
+    soc_series = pd.Series(data=data, index=index)
+
+    storage = "storage_01"
+
+    # Add the SoC time series to the result JSON nested-dict
+    dict_values = {ENERGY_STORAGE: {storage: {TIMESERIES_SOC: soc_series}}}
+
+    # Test the function with the present case
+    with caplog.at_level(logging.WARNING):
+        E4.verify_state_of_charge(dict_values=dict_values)
+    assert (
+        f"SoC of {storage} in this time-step is less than 0. This is a physically impossible value!"
+        in caplog.text
+    ), f"A WARNING message is not logged even though the SoC values are all below 0."
+
+
+def test_verify_state_of_charge_soc_above_zero(caplog):
+    """Test case: The SoC value are all above 0.00 and thus, physically infeasible
+    Test to check if the WARNING log messages are logged"""
+
+    # Create a list of floats greater than 1.00
+    list_floats = [1.23, 1.0001, 2.05]
+    data = np.array(list_floats)
+
+    # Generate a date-time index
+    index = pd.date_range(start="1/1/2018", periods=3, tz="Asia/Kolkata")
+
+    # Create a SoC time series from above floats and date-time data
+    soc_series = pd.Series(data=data, index=index)
+
+    storage = "storage_01"
+
+    # Add the SoC time series to the result JSON nested-dict
+    dict_values = {ENERGY_STORAGE: {storage: {TIMESERIES_SOC: soc_series}}}
+
+    # Test the function with the present case
+    with caplog.at_level(logging.WARNING):
+        E4.verify_state_of_charge(dict_values=dict_values)
+    assert (
+        f"SoC of {storage} in this time-step is greater than 1. This is a physically impossible value!"
```

Comment:
```suggestion
        f"greater than 1. This is a physically impossible value!"
```
This is how you would need to change the assertions for the above proposed version with only one error message (not one per timestep) to work.

Reply:
Of course this adaption would be necessary for the other assertions as well.

Reply:
Done @smartie2076 

Fixed the function and the tests according to your suggestion